### PR TITLE
Bot and strategy base API

### PR DESCRIPTION
**Pull request by sergeypospelov**
TODO: update docs

### CODE REVIEW

File: bot-service/src/main/kotlin/backend/bot/service/SimpleBotService.kt [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import backend.tinkoff.account.TinkoffActualAccount
+import backend.tinkoff.account.TinkoffVirtualAccountFactory
+import backend.tinkoff.model.Currency
+import backend.tinkoff.model.Quotation
+import java.util.concurrent.atomic.AtomicInteger
+
+class SimpleBotService(
+    configure: Configuration.() -> Unit
+) : BotService {
+    private val virtualAccoutnFactory: TinkoffVirtualAccountFactory
```
Comment (SBOne-Kenobi) [inline bot-service/src/main/kotlin/backend/bot/service/SimpleBotService.kt:22]: account

File: bot-service/src/main/kotlin/backend/bot/clusters/SimpleCluster.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    override fun deploy(container: StrategyContainer, name: BotName, parameters: Parameters, tinkoffAccount: TinkoffAccount): Result<BotUid> {
+        val configuration = Configuration(
+            parameters,
+            tinkoffAccount
+        )
+        val res = container.start(configuration)
+
+        return if (res) {
+            val uid = botNumberer.incrementAndGet()
```
Comment (SBOne-Kenobi) [inline bot-service/src/main/kotlin/backend/bot/clusters/SimpleCluster.kt:42]: Maybe later think about inheriting uid from account or getting it from database

File: bot-service/src/main/kotlin/backend/bot/Api.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    fun startBot(strategyUid: StrategyUid, name: BotName, parameters: Parameters): Result<BotUid>
+    fun stopBot(uid: BotUid): Boolean
+}
+
+interface BotCluster {
+    fun activeBots(): List<BotUid>
+    fun getBot(uid: BotUid): Result<Bot>
+    fun deploy(container: StrategyContainer, name: BotName, parameters: Parameters, tinkoffAccount: TinkoffAccount): Result<BotUid>
+    fun stopBot(uid: BotUid): Boolean
+}
```
Comment (SBOne-Kenobi) [inline bot-service/src/main/kotlin/backend/bot/Api.kt:32]: In the future, it's preferable to make functions with an async result or just with a suspend modificator

File: strategy-service/src/main/kotlin/backend/strategy/strategies/megastupid/MegaStupidStrategyContainer.kt [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+class MegaStupidStrategyContainer : StrategyContainer {
+    private val executor = Executors.newSingleThreadExecutor()
+    private val dispatcher = executor.asCoroutineDispatcher()
+    private val scope = CoroutineScope(dispatcher)
+    private lateinit var job: Job
+
+
+    override fun start(configuration: Configuration): Boolean {
+        job = scope.launch {
+            strategy(configuration.tinkoffAccount)
```
Comment (SBOne-Kenobi) [inline strategy-service/src/main/kotlin/backend/strategy/strategies/megastupid/MegaStupidStrategyContainer.kt:29]: Think about errors and exceptions. Need to unregister or something like that.

File: strategy-service/src/main/kotlin/backend/strategy/strategies/stupid/StupidStrategyContainer.kt [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+class StupidStrategyContainer : StrategyContainer {
+    private val executor = Executors.newSingleThreadExecutor()
+    private val dispatcher = executor.asCoroutineDispatcher()
+    private val scope = CoroutineScope(dispatcher)
+    private lateinit var job: Job
+
+
+    override fun start(configuration: Configuration): Boolean {
+        job = scope.launch {
+            strategy(configuration.tinkoffAccount)
```
Comment (SBOne-Kenobi) [inline strategy-service/src/main/kotlin/backend/strategy/strategies/stupid/StupidStrategyContainer.kt:29]: Same

File: tinkoff-wrapper/src/main/kotlin/backend/tinkoff/account/TinkoffAccount.kt [hunk 6]
```diff
@@ -24,4 +25,5 @@ interface TinkoffAccount {
     fun getOpenOrders(): Result<List<OrderState>>
 
     fun getPositions(): Result<PositionsResponse>
+    fun getLastPrice(figi: String): Result<Quotation>
```
Comment (IdeaSeeker) [inline tinkoff-wrapper/src/main/kotlin/backend/tinkoff/account/TinkoffAccount.kt:28]:
```suggestion

    fun getLastPrice(figi: String): Result<Quotation>
```

**Review by IdeaSeeker — APPROVED** [review]

**Pull request merged**